### ISSUE TITLE
Amazon API results give overspecific (made up) day and month for older books

### ISSUE THREAD

**Issue by hornc** [internetarchive/openlibrary#2334]
### Is your feature request related to a problem? Please describe.

Older books sourced from Amazon metadata may have over specific dates from earlier than a particular year.

Needs investigation, but Amazon sourced dates have a particular format, with day, month, year, Beyond a certain point, the API seems to use the current day and month because their record does not actually have one -- I'll need to confirm this, and check how much of our exisiting data is affected, but I wanted to create the issue to remind myself to investigate further.

It means the publication year is good, but month and day portions are likely junk.

3 parts:

- [x] Investigate and confirm the issue, id the period where specific dates are good vs. bad
- [ ] Identify existing records affected (matching criteria to locate them in dump files and count of affected records)
- [ ] Adjust import code to process these dates appropriately
- [ ] Clean up existing records

**Comment by hornc** [comment]
Confirmed the issue -- for an Amazon API response for this book http://amazon.com/dp/B001PDNXXC

The direct API response has:
`<PublicationDate>1947</PublicationDate>`

and the converted JSON we use in our import is 
`publish_date: "Aug 26, 1947"`

Current Open Library server date when I performed this lookup is Aug 26. 

The source of this problem is in the simple Amazon API tool we use: https://github.com/yoavaviram/python-amazon-simple-product-api/blob/84c16f519e5a349a30669bc4218b9325ca37f651/amazon/api.py#L490  where the date string is converted into a `date` object using python's `dateutil.parser`, which adds month and day:
```
>>> import dateutil.parser
>>> dateutil.parser.parse('1947')
datetime.datetime(1947, 8, 26, 0, 0)
```

So this is not a problem with Amazon's API, but how we're using the data returned from it. 
I believe some of the more modern publication dates from Amazon have day and month, and we should use + display those if present.
